### ISSUE TITLE
Unable to set values for git config when global is symlinked

### ISSUE THREAD

**Issue by sentientmonkey** [libgit2/libgit2#4214]
You are opening a _bug report_ against the libgit2 project.  If you have a
question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2.  Please fill out the
reproduction steps (below) and delete this introductory paragraph.  Thanks!

### Reproduction steps
1. Download repo via https://gist.github.com/sentientmonkey/302a1ccceff64e254337eeb78536352c
2. git init a local repo (via `git init .`)
3. Symlink global .gitconfig (i.e. "/Users/user/.gitconfig -> /Users/user/workspace/dotfiles/.gitconfig")
4. Build and run example code ```gcc -o main -lgit2 main.c && ./main```

### Expected behavior
Example should print "foo.bar is baz"

### Actual behavior
```Error -1 set_string - failed to stat '/Users/user/.gitconfig/Users/user/workspace/dotfiles/.gitconfig': Not a directory```

### Version of libgit2 (release number or SHA1)
0.25.1

### Operating system(s) tested
macOS Sierra 10.12.4 (16E195)

**Comment by pks-t** [comment]
I'm not able to reproduce this. I've symlinked by configuration since forever and libgit2 has always been able to correctly modify the symlinked file. So I guess this is something specific to either your setup or macOS. Could you maybe get a backtrace where the error is initially raised?

**Comment by sentientmonkey** [comment]
Was able to reproduce this in a docker container from the latest stable release.

(see Dockerfile below)
https://gist.github.com/sentientmonkey/65f31a32336724b30f784e11f1e7b37e

But, in running this through gdb looks like this was fixed in the latest HEAD from...

https://github.com/libgit2/libgit2/commit/86a8cd9f6a039889801b5bec865a4bc3deb30f47

Once I made the same change locally, it's able to resolve the symlink properly. I'll go ahead and close out this issue.
